lib/spack/spack/hooks/__init__.py

```diff
@@ -37,6 +37,10 @@ def all_hook_modules():
         path = os.path.join(spack.paths.hooks_path, name) + ".py"
         mod = simp.load_source(mod_name, path)
         modules.append(mod)
+        if name == 'write_install_manifest':
+            last_mod = mod
+
+    modules.sort(key=lambda x: x is last_mod)
```

Comment:
I think a comment here would be more appropriate than refactoring. I'll add one.

Reply:
I feel this is too complex, and you still have to combine it with the calculation of `last_mod` to figure out you want to place `write_install_manifest` accordingly, and on top of that sorting can still generate confusion because you are doing it to place one thing at the beginning or end of the list. If code has to be complex then a comment is the right way to go, but if it can be simple then I don't think a comment is sufficient.